catkin_ws/src/sim/src/unity_bridge.py

```diff
@@ -38,124 +105,169 @@ def cb_unity_state(msg):
-    q_dvlnominal_dvl_z = rospy.get_param("q_dvlnominal_dvl_z")
-    q_dvlnominal_dvl = np.quaternion(
-        q_dvlnominal_dvl_w, q_dvlnominal_dvl_x, q_dvlnominal_dvl_y, q_dvlnominal_dvl_z
+    q_dvlnominalup_dvlup_w = rospy.get_param("q_dvlnominalup_dvlup_w")
+    q_dvlnominalup_dvlup_x = rospy.get_param("q_dvlnominalup_dvlup_x")
+    q_dvlnominalup_dvlup_y = rospy.get_param("q_dvlnominalup_dvlup_y")
+    q_dvlnominalup_dvlup_z = rospy.get_param("q_dvlnominalup_dvlup_z")
+    q_dvlnominalup_dvlup = np.quaternion(
+        q_dvlnominalup_dvlup_w, q_dvlnominalup_dvlup_x, q_dvlnominalup_dvlup_y, q_dvlnominalup_dvlup_z
     )
 
     auv_dvl_offset_x = rospy.get_param("auv_dvl_offset_x")
     auv_dvl_offset_y = rospy.get_param("auv_dvl_offset_y")
     auv_dvl_offset_z = rospy.get_param("auv_dvl_offset_z")
 
-    q_imunominal_imu_w = rospy.get_param("q_imunominal_imu_w")
-    q_imunominal_imu_x = rospy.get_param("q_imunominal_imu_x")
-    q_imunominal_imu_y = rospy.get_param("q_imunominal_imu_y")
-    q_imunominal_imu_z = rospy.get_param("q_imunominal_imu_z")
+    q_imunominalup_imuup_w = rospy.get_param("q_imunominalup_imuup_w")
+    q_imunominalup_imuup_x = rospy.get_param("q_imunominalup_imuup_x")
+    q_imunominalup_imuup_y = rospy.get_param("q_imunominalup_imuup_y")
+    q_imunominalup_imuup_z = rospy.get_param("q_imunominalup_imuup_z")
 
-    q_imunominal_imu = np.quaternion(
-        q_imunominal_imu_w, q_imunominal_imu_x, q_imunominal_imu_y, q_imunominal_imu_z
+
+    q_imunominalup_imuup = np.quaternion(
+        q_imunominalup_imuup_w, q_imunominalup_imuup_x, q_imunominalup_imuup_y, q_imunominalup_imuup_z
     )
 
+    q_imunominaldown_imunominalup = np.quaternion(0,1,0,0)
+    q_imunominalup_dvlnominalup = np.quaternion(1,0,0,0)
+
     # REFERENCE FRAME DEFINITIONS
-    random_vector = np.random.rand(4)
-    random_vector = random_vector / np.linalg.norm(random_vector)
-    # q_NWU_dvlref = np.quaternion(random_vector[0],random_vector[1],random_vector[2],random_vector[3])
-    q_NWU_dvlref = np.quaternion(0, 1, 0, 0)
-    q_imunominal_auv = np.quaternion(0, 1, 0, 0)
-    q_dvlnominal_auv = np.quaternion(0, 1, 0, 0)
-    q_NWU_NED = np.quaternion(0, 1, 0, 0)
-    q_dvl_auv = q_dvlnominal_dvl.conjugate() * q_dvlnominal_auv
+    q_ENU_ESD = np.quaternion(0, 1, 0, 0)
+
+
+    q_imunominaldown_dvlnominalup = np.quaternion(0,1,0,0)
 
-    # TODO: ADD DVL VELOCITY
 
     pub_dvl_sensor = rospy.Publisher(
-        "/sensors/dvl/pose", DeadReckonReport, queue_size=1
+        "/sensors/dvl/twist", TwistWithCovarianceStamped, queue_size=1
     )
-    pub_depth_sensor = rospy.Publisher("/sensors/depth/z", Float64, queue_size=1)
-    pub_imu_quat_sensor = rospy.Publisher(
-        "/sensors/imu/quaternion", SbgEkfQuat, queue_size=1
-    )
-    pub_imu_data_sensor = rospy.Publisher(
-        "/sensors/imu/angular_velocity", SbgImuData, queue_size=1
+    pub_depth_sensor = rospy.Publisher("/sensors/depth/pose", PoseWithCovarianceStamped, queue_size=1)
```

Comment:
the publishers (topics and types) here don't match the sensor_status subscribers (topics and types)

Reply:
which ones? depth sensor get replublished

Reply:
oh wait i see the issue

Reply:
yeah. You need to control shift F each topic, change to this new one, but also (maybe) change the logic for the section that was using that topic